qemu/target-i386/unicorn.c

```diff
@@ -260,22 +260,22 @@ int x86_reg_read(struct uc_struct *uc, unsigned int regid, void *value)
                     *(int16_t *)value = READ_WORD(X86_CPU(uc, mycpu)->env.eip);
                     break;
                 case UC_X86_REG_CS:
-                    *(int32_t *)value = X86_CPU(uc, mycpu)->env.segs[R_CS].base;
+                    *(int32_t *)value = (uint16_t)X86_CPU(uc, mycpu)->env.segs[R_CS].selector;
```

Comment:
yes

On 2/4/2016 6:54 PM, Nguyen Anh Quynh wrote:
> In qemu/target-i386/unicorn.c <https://github.com/unicorn-engine/unicorn/pull/422#discussion_r51971898>:
> 
>> @@ -260,22 +260,22 @@ int x86_reg_read(struct uc_struct *uc, unsigned int regid, void *value)
>>                      *(int16_t *)value = READ_WORD(X86_CPU(uc, mycpu)->env.eip);
>>                      break;
>>                  case UC_X86_REG_CS:
>> -                    *(int32_t *)value = X86_CPU(uc, mycpu)->env.segs[R_CS].base;
>> +                    *(int32_t *)value = (uint16_t)X86_CPU(uc, mycpu)->env.segs[R_CS].selector;
> 
> on the left side, don't you want to cast to pointer of *uint16_t* here, rather than int32_t?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/unicorn-engine/unicorn/pull/422/files#r51971898>.
>